clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -915,6 +915,41 @@ def ExtraFuncAttr : CIR_Attr<"ExtraFuncAttributes", "extra"> {
   // Printing and parsing also available in CIRDialect.cpp
 }
 
+def VK_Default : I32EnumAttrCase<"Default", 1, "default">; 
+def VK_Hidden : I32EnumAttrCase<"Hidden", 2, "hidden">;
+def VK_Protected : I32EnumAttrCase<"Protected", 3, "protected">;
+
+def VisibilityKind : I32EnumAttr<"VisibilityKind", "visibilityKind", [
+  VK_Default, VK_Hidden, VK_Protected
+]> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def VisibilityAttr : CIR_Attr<"Visibility", "visibility"> {
```

Comment:
Just occured to me that the generated names are getting very confusing if we use `VisibilityAttr`. Perhaps it should be named `CIRVisibilityAttr`, or some other better name, so that the generated methods can be different enough from `setVisibility`.